examples/pump-app/linux/PumpManager.cpp

```diff
@@ -0,0 +1,296 @@
+{
+    mAutoStartDuration = aDurationInSecs;
+}
+
+bool PumpManager::InitiateAction(int32_t aActor, Action_t aAction)
+{
+    bool action_initiated = false;
+    State_t new_state;
+
+    // Initiate Start/Stop Action only when the previous one is complete.
+    if (mState == kState_StartCompleted && aAction == STOP_ACTION)
+    {
+        action_initiated = true;
+
+        new_state = kState_StartInitiated;
+    }
+    else if (mState == kState_StopCompleted && aAction == START_ACTION)
+    {
+        action_initiated = true;
+
+        new_state = kState_StartInitiated;
+    }
+
+    if (action_initiated)
+    {
+        if (mAutoStartTimerArmed && new_state == kState_StartInitiated)
+        {
+            // If auto start timer has been armed and someone initiates start,
+            // cancel the timer and continue as normal.
+            mAutoStartTimerArmed = false;
+
+            CancelTimer();
+        }
+
+        PumpTimer(ACTUATOR_MOVEMENT_PERIOS_MS);
+
+        // Since the timer started successfully, update the state and trigger callback
+        mState = new_state;
+
+        if (mActionInitiated_CB)
+        {
+            mActionInitiated_CB(aAction, aActor);
+        }
+    }
+
+    return action_initiated;
+}
+
+void PumpManager::PumpTimer(uint32_t aTimeoutMs) {}
+
+void PumpManager::CancelTimer(void) {}
+
+void PumpManager::AutoRestartTimerEventHandler(AppEvent * aEvent)
+{
+    PumpManager * pump = static_cast<PumpManager *>(aEvent->PumpStateEvent.Context);
+    int32_t actor      = 0;
+
+    // Make sure auto start timer is still armed.
+    if (!pump->mAutoStartTimerArmed)
+    {
+        return;
+    }
+
+    pump->mAutoStartTimerArmed = false;
+
+    ChipLogProgress(Zcl, "Auto Re-Start has been triggered!");
+
+    pump->InitiateAction(actor, START_ACTION);
+}
+
+void PumpManager::ActuatorMovementTimerEventHandler(AppEvent * aEvent)
+{
+    Action_t actionCompleted = INVALID_ACTION;
+
+    PumpManager * pump = static_cast<PumpManager *>(aEvent->PumpStateEvent.Context);
+
+    if (pump->mState == kState_StartInitiated)
+    {
+        pump->mState    = kState_StartCompleted;
+        actionCompleted = START_ACTION;
+    }
+    else if (pump->mState == kState_StopInitiated)
+    {
+        pump->mState    = kState_StopCompleted;
+        actionCompleted = STOP_ACTION;
+    }
+
+    if (actionCompleted != INVALID_ACTION)
+    {
+        if (pump->mActionCompleted_CB)
+        {
+            pump->mActionCompleted_CB(actionCompleted);
+        }
+
+        if (pump->mAutoRestart && actionCompleted == STOP_ACTION)
+        {
+            // Start the timer for auto restart
+            pump->PumpTimer(pump->mAutoStartDuration * 1000);
+
+            pump->mAutoStartTimerArmed = true;
+
+            ChipLogProgress(Zcl, "Auto Re-start enabled. Will be triggered in %u seconds", pump->mAutoStartDuration);
+        }
+    }
+}
+
+int16_t PumpManager::GetMaxPressure()
```

Comment:
Why do you need getter functions with hardcoded values ? If it's absolutely necessary, can you please add a comment why it is inside of the code and replace all of the magic numbers with defines or static const expression ?

Reply:
Primarily because they need to be "connected" to some code, in future PR's, which will supply the PumpManager with these default values. Currently we have just added them for completeness.

BTW. How can you supply the Attributes with default values using other mechanisms? I have been looking into the .zap file for the PCC Cluster, but is it supported by the Zaptool yet?